src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -4815,6 +4815,19 @@ ProjectionNames QueryAnalyzer::resolveMatcher(QueryTreeNodePtr & matcher_node, I
         }
     }
 
+    if (!scope.expressions_in_resolve_process_stack.hasAggregateFunction())
+    {
+        for (auto & [node, _] : matched_expression_nodes_with_names)
+        {
+            auto it = scope.nullable_group_by_keys.find(node);
+            if (it != scope.nullable_group_by_keys.end())
+            {
+                node = it->node->clone();
+                node->convertToNullable();
+            }
+        }
+    }
```

Comment:
Hmm. Interesting case.
Is it about something like `SELECT ... GROUP BY ... ORDER BY *`?

Reply:
Should be the same, we resolve `*` using the same method not depending on it's source as usual identifiers. Added tests for it